modules/portfolio/lib/portfolio-data.service.ts

```diff
@@ -92,35 +92,50 @@ export class PortfolioDataService {
         const blockNumber = parseInt(block.number);
         const poolShares = [];
 
-        for (const sharesOwned of user.sharesOwned ?? []) {
-            const pool = pools.find((pool) => pool.id === sharesOwned.poolId.id);
-
-            if (!pool) {
-                continue;
+        const poolIds = [];
```

Comment:
Type your variables whenever possible, the `poolShares` above is not typed because the type is ensured in the return statement. I could push a boolean into your `poolIds` and it wouldn't complain

Reply:
Gotcha. Done.